meta2v2/meta2_utils.c

```diff
@@ -1181,28 +1232,65 @@ m2db_delete_alias(struct sqlx_sqlite3_s *sq3, gint64 max_versions,
 			max_versions, ALIASES_get_deleted(alias),
 			oio_url_has(url, OIOURL_VERSION), oio_url_get(url, OIOURL_VERSION));
 
+	gboolean add_delete_marker = FALSE;
 	if (VERSIONS_DISABLED(max_versions) || VERSIONS_SUSPENDED(max_versions) ||
 			oio_url_has(url, OIOURL_VERSION) || ALIASES_get_deleted(alias)) {
-		err = _real_delete_and_save_deleted_beans(sq3, beans, alias, header,
-				cb, u0);
+		if (delete_marker) {
+			if (VERSIONS_DISABLED(max_versions) ||
+					VERSIONS_SUSPENDED(max_versions)) {
+				err = NEWERROR(CODE_BAD_REQUEST,
```

Comment:
The `BADREQ` macro might help you to keep the lines short enough to avoid folding.